### DISCUSSION TITLE
How to enable SSR per page, not just of `App`?

### DISCUSSION THREAD

**Discussion by MarioIshac** [vitejs/vite-plugin-react#381] [Q&A] [answered]
For `playground/ssr-react`, when running `server.js` and requesting `/env`, I get this:

```
<!doctype html>
<html lang="en">
  <head>
    <script type="module">
import RefreshRuntime from "/@react-refresh"
RefreshRuntime.injectIntoGlobalHook(window)
window.$RefreshReg$ = () => {}
window.$RefreshSig$ = () => (type) => type
window.__vite_plugin_react_preamble_installed__ = true
</script>

    <script type="module" src="/@vite/client"></script>

    <meta charset="UTF-8" />
    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
    <title>Vite App</title>
  </head>
  <body>
    <div id="app"><nav><ul><li><a href="/about">About</a></li><li><a href="/env">Env</a></li><li><a href="/">Home</a></li></ul></nav><h1>API_KEY_qwertyuiop</h1></div>
    <script type="module" src="/src/entry-client.jsx"></script>
  </body>
</html>
```
As you can see inside the `div` of the `body`, only the `App` layout is rendered, not the `Env` I requested. In the browser, I then see that the client side router takes over and renders `Env` client-side.

Is it possible to have SSR per page with the entire component tree of that page rendered upfront, not just `App`?

**Comment by hi-ogawa** [comment, marked as answer]
`<h1>API_KEY_qwertyuiop</h1>` is what's rendered by `pages/Env.jsx` so it looks working though. What do you expect to see instead?

https://github.com/vitejs/vite-plugin-react/blob/b61592a6ffb9e36c3ffb887c30baa29b6df4f53c/playground/ssr-react/server.js#L10

https://github.com/vitejs/vite-plugin-react/blob/b61592a6ffb9e36c3ffb887c30baa29b6df4f53c/playground/ssr-react/src/pages/Env.jsx#L1-L7

**Reply by MarioIshac** [reply]
Apologies, my mistake! I was looking for "default message here" like I see in browser, but I realize that some process afterwards is replacing `<h1>API_KEY_qwertyuiop</h1>` with that right after page load (is that the hydration?), and that the page-specific SSR is still happening. Thank you for the quick clarification.

**Reply by hi-ogawa** [reply]
Yeah, this isn't a best demo as it's expected to have hydration error, but I guess it's serving as test cases.
FYI, Remix team also has examples based on Vite on their repo https://github.com/remix-run/react-router/tree/main/examples if you haven't looked already.
